matter_server/server/ota/provider.py

```diff
@@ -326,6 +302,55 @@ async def download_update(self, update_desc: dict) -> None:
             )
             raise UpdateError("Fetching software version failed") from err
 
+        return file_path
+
+    async def fetch_update(self, update_desc: dict) -> None:
+        """Fetch update file from OTA URL."""
+        url = update_desc["otaUrl"]
+        parsed_url = urlparse(url)
+        file_name = unquote(Path(parsed_url.path).name)
+
+        loop = asyncio.get_running_loop()
+
+        checksum_alg = None
+        if (
+            "otaChecksum" in update_desc
+            and "otaChecksumType" in update_desc
+            and update_desc["otaChecksumType"] in CHECHKSUM_TYPES
+        ):
+            checksum_alg = hashlib.new(CHECHKSUM_TYPES[update_desc["otaChecksumType"]])
+        else:
+            LOGGER.warning(
+                "No OTA checksum type or not supported, OTA will not be checked."
+            )
+
+        if parsed_url.scheme in ["http", "https"]:
+            file_path = await self._download_update(url, checksum_alg)
+        elif parsed_url.scheme in ["file"]:
+            file_path = self._ota_provider_base_dir / Path(parsed_url.path[1:])
+            if not file_path.exists():
+                logging.warning("Local update file not found: %s", file_path)
+                raise UpdateError("Local update file not found")
+            if checksum_alg:
+                checksum_alg.update(
+                    await loop.run_in_executor(None, file_path.read_bytes)
```

Comment:
We can use `asyncio.to_thread` now. It does the same thing.

Reply:
Implemented with #886.